okio/src/main/java/okio/AsyncTimeout.java

```diff
@@ -151,6 +130,19 @@ private long remainingNanos(long now) {
   protected void timedOut() {
   }
 
+  @Override
+  public final long getDelay(TimeUnit unit) {
+    return unit.convert(timeoutAt - System.nanoTime(), TimeUnit.NANOSECONDS);
+  }
+
+  @Override
+  public final int compareTo(Delayed that) {
+    if (that instanceof AsyncTimeout) {
+      return (int) (timeoutAt - ((AsyncTimeout) that).timeoutAt);
+    }
+    return (int) (getDelay(TimeUnit.NANOSECONDS) - that.getDelay(TimeUnit.NANOSECONDS));
```

Comment:
Not sure how I feel about this. I think it might be simplest just to throw a ClassCastException in this case.